packages/near-membrane-base/src/membrane.ts

```diff
@@ -250,15 +246,21 @@ export function init(
     // to the callable functions used to coordinate work between the sides
     // of the membrane.
     // TODO: do we need to pass more info into instrumentation hooks?
-    function instrumentCallableWrapper<T extends (...args: any[]) => any>(fn: T, label: string): T {
+    // prettier-ignore
+    function instrumentCallableWrapper<T extends (...args: any[]) => any>(
+        fn: T,
+        activityName: string,
+        crossingDirection: string): T {
         if (instrumentation) {
             return <T>function instrumentedFn(this: any, ...args: any[]): any {
                 let result;
-                instrumentation.start(label);
+                const activity = instrumentation.startActivity(activityName, { crossingDirection });
                 try {
                     result = ReflectApply(fn, this, args);
+                } catch (ex) {
+                    activity.error(ex as any);
```

Comment:
TypeScript makes the catch variable type to `unknown`. Casting is needed.

Reply:
you can even do `catch (e: any)`. I believe we usually use `e` for the error and not `ex`.

Reply:
We should rethrow the error. At the moment it's getting swallowed.

Reply:
good tips. Thanks!